### PR TITLE
KFSPTS-16483: Additional CGB Invoice placeholders for Award and Award Account.

### PR DESCRIPTION
**Pull request by jhulslander**
This is a re-work of the original change set to set the purchase order placeholder differently depending on the invoice option.

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java [hunk 1]
```diff
@@ -67,6 +73,93 @@
...
+    protected String findPurchaseOrderNumberForInvoiceOptionAward(Award award) {
+        String purchaseOrderNumber = StringUtils.EMPTY;
+        AwardExtendedAttribute awardExtension = (AwardExtendedAttribute) award.getExtension();
+        if (ObjectUtils.isNotNull(awardExtension)) {
+            purchaseOrderNumber = awardExtension.getPurchaseOrderNumber();
+        }
+        return purchaseOrderNumber;
+    }
+
+    protected String findPurchaseOrderNumberForInvoiceOptionAccoiunt(ContractsGrantsInvoiceDocument document) {
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java]: Minor typo in method name; should be changed from "...Accoiunt" to "...Account"

File: src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java [hunk 2]
```diff
@@ -67,6 +73,93 @@
...
+    protected AwardAccountExtendedAttribute findAwardAccountExtendedAttribute(String proposalNumber, String chartOfAccountsCode, String accountNumber) {
+        Map<String, Object> primaryKeys = new HashMap<String, Object>();
+        primaryKeys.put(KFSPropertyConstants.PROPOSAL_NUMBER, proposalNumber);
+        primaryKeys.put(KFSPropertyConstants.CHART_OF_ACCOUNTS_CODE, chartOfAccountsCode);
+        primaryKeys.put(KFSPropertyConstants.ACCOUNT_NUMBER, accountNumber);
+        AwardAccountExtendedAttribute awardAccountExtension = businessObjectService.findByPrimaryKey(AwardAccountExtendedAttribute.class, primaryKeys);
+        return awardAccountExtension;
+    }
+
+    protected String findPurchaseOrderNymberForInvoiceOptionContractControlAndScheduled(ContractsGrantsInvoiceDocument document) {
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java]: Minor typo in method name; should be changed from "...OrderNymber..." to "...OrderNumber..."

File: src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java (hunk 1 above)
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java]: ok

File: src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java (hunk 2 above)
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java]: ok

**Comment by jhulslander** [comment]
I've made some changes, this is ready for review again.

**Pull request merged**